tests/ui/macros/issue-112342-1.rs

```diff
@@ -0,0 +1,33 @@
+// same as #95267, ignore doc comment although it's a bug.
+
+macro_rules! m1 {
+    (
+        $(
+            ///
+        )*
+    ) => {};
+}
+
+m1! {} //~ERROR unexpected end of macro invocation
```

Comment:
even if it should fail, the error looks wrong
```
   | ^^^^^^ missing tokens in macro arguments
   |
   = note: while trying to match sequence start
```

Reply:
I haven't looked at this in detail (I have some familiarity with the code but it's been a while) but I don't think this is the correct solution if it leads to this behavior.

Reply:
> I would expect this to pass

Maybe not, `macro_rules! m { ( $()* ) => {} }` will throw error.

> the error looks wrong

Yes, you are right, naked break is not good for this situation, I will change `break` to `continue`, it will emit error same as:

```rs
macro_rules! m {
    (
        $(
            ///
        )?
    ) => {};
}
```

As for the error throwing `missing tokens in macro arguments`, I think it would be better to fix it in a separate PR.